helix-term/src/ui/picker.rs

```diff
@@ -715,28 +721,33 @@ impl<T: Item + Send + 'static> Component for DynamicPicker<T> {
     }
 
     fn handle_event(&mut self, event: &Event, cx: &mut Context) -> EventResult {
-        let prev_query = self.file_picker.picker.prompt.line().to_owned();
         let event_result = self.file_picker.handle_event(event, cx);
         let current_query = self.file_picker.picker.prompt.line();
 
-        if *current_query == prev_query || matches!(event_result, EventResult::Ignored(_)) {
+        if !matches!(event, Event::IdleTimeout) || self.query == *current_query {
             return event_result;
         }
 
+        self.query.clone_from(current_query);
+
         let new_options = (self.query_callback)(current_query.to_owned(), cx.editor);
 
         cx.jobs.callback(async move {
             let new_options = new_options.await?;
-            let callback: crate::job::Callback = Box::new(move |_editor, compositor| {
+            let callback: crate::job::Callback = Box::new(move |editor, compositor| {
                 // Wrapping of pickers in overlay is done outside the picker code,
                 // so this is fragile and will break if wrapped in some other widget.
-                let picker = match compositor.find_id::<Overlay<DynamicPicker<T>>>(Self::ID) {
-                    Some(overlay) => &mut overlay.content.file_picker.picker,
+                let file_picker = match compositor.find_id::<Overlay<DynamicPicker<T>>>(Self::ID) {
+                    Some(overlay) => &mut overlay.content.file_picker,
                     None => return,
                 };
-                picker.options = new_options;
-                picker.cursor = 0;
-                picker.force_score();
+                file_picker.picker.options = new_options;
+                file_picker.picker.cursor = 0;
+                file_picker.picker.force_score();
+                // This callback is triggered on the idle timeout, so we simulate the
+                // file-picker's handle_idle_timeout behavior by highlighting the
+                // currently previewed file.
+                file_picker.highlight_current_file(editor);
```

Comment:
Is this needed here? Isn't the highlighting already handled by calling `file_picker.handle_event()` at the beginning of the function, which calls `handle_idle_timeout()`?

https://github.com/helix-editor/helix/blob/26f21da531179ccc77e70b73e32a1b458c4e225b/helix-term/src/ui/picker.rs#L289-L292

Reply:
If the new symbols response triggered by the idle timeout selects a new file then it won't be highlighted. I just noticed that it doesn't work anyways so I'll remove this 😅

Reply:
(for example searching for something specific like "flip" in the helix repo)